conda_build/os_utils/external.py

```diff
@@ -58,8 +58,19 @@ def find_executable(executable, prefix=None, all_matches=False):
     return result or all_matches_found
 
 
-def find_preferably_prefixed_executable(executable, build_prefix=None, all_matches=False):
-    found = find_executable('*' + executable, build_prefix, all_matches)
+def find_preferably_prefixed_executable(executable, build_prefix=None, all_matches=False,
+        host_subdir=None):
+    if host_subdir == "osx-64":
+        exe_prefix = "x86_64-*"
+    elif host_subdir == "osx-arm64":
+        exe_prefix = "arm64-*"
+    elif not host_subdir:
+        exe_prefix = "*"
+    else:
+        import warnings
+        warnings.warn('unknown host_subdir' % host_subdir, UserWarning)
```

Comment:
Wouldn't that imply that something unexpected happened and we should fail instead of warn?

Also small typo:

```suggestion
        warnings.warn('unknown host_subdir %s' % host_subdir, UserWarning)
```